src/emitEventsInBatches.js

```diff
@@ -0,0 +1,48 @@
+const chunk = require('lodash.chunk');
+const { enrichMeta, partitionKey } = require('./enrich')
+
+const MAX_RECORDS = 500; 
+
+const emitEvents = async (kinesis, events, config, retries) => {
+  const records = events.map(event => {
+    const enrichedEvent = enrichMeta(event, config.appName, config.ipv4);
```

Comment:
A small question here: in case of a retry, events will already have beeen enriched, I am guessing the operation just does nothing for an event that was previously enriched, but might be worth double checking?

Alternatively, it would be possible to separate the enriching part out of the actual call + retry.

Reply:
It updates the `createdAt` attribute to the latest datetime for the failed events, it might only be a few (m)secs difference but just trying to keep it real here.

Reply:
Good idea but it won't actually update it 😅 
At the [update location](https://github.com/babbel/miza-kinesis/pull/179/files#diff-232774a86743748ff11ce5542109915282149b877556c4dd5defd20eb24d45cbR17), the newly generated meta is the object that gets assigned too, so for any keys in common between `event` and the newly generated meta, the one in `event` will overwrite the one in the generated meta.
So the first generated `createdAt` will be assigned after the second enriching.

Reply:
yeah, the `Object.assign` 😀 

I'll try to extract the enriching part 👍

Reply:
I updated the code to break the events into chunks of 500 events first, then enrich each chunk and transform them into records that can be emitted. On partial failure of each chunk, it will be retried immediately so there won't be big difference in created at attribute so I didn't make changes in `enrichMeta` function.

Reply:
and also we shouldn't be updating `createdAt` but we can update the `meta.createdAt` this is important for offline events and buffered events that get emitted when the user is online.

Reply:
Agreed, though because of the way the `Object.assigns` are written, both `createAt` and `meta.createdAt` will never be updated if they already exist.
From what I see, in the current version `meta.createdAt` will not be updated, I think I read from your messages that that should be alright?

Reply:
yeah, meta.createdAt will not be updated and it should be okay!